From 6aebb3b87e61b1ee937ce5cb6b9550a07a0dcf36 Mon Sep 17 00:00:00 2001
From: Khanh Tran <khanh.tran.te@renesas.com>
Date: Tue, 20 Apr 2021 16:50:07 +0700
Subject: [PATCH] rpmsg: virtio: fix get buffer size return If the length in
 buffer descriptor is 0 or less then the RPMsg header, return 0

---
 lib/rpmsg/rpmsg_virtio.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/rpmsg/rpmsg_virtio.c b/lib/rpmsg/rpmsg_virtio.c
index 1958165..ca8db6a 100644
--- a/lib/rpmsg/rpmsg_virtio.c
+++ b/lib/rpmsg/rpmsg_virtio.c
@@ -271,6 +271,9 @@ static int _rpmsg_virtio_get_buffer_size(struct rpmsg_virtio_device *rvdev)
 		length =
 		    (int)virtqueue_get_desc_size(rvdev->svq) -
 		    sizeof(struct rpmsg_hdr);
+		if (length < 0) {
+		    length = 0;
+		}
 	}
 #endif /*!VIRTIO_MASTER_ONLY*/
 
-- 
2.7.4

